relayer/strategies.go

```diff
@@ -186,9 +199,10 @@ func applyChannelFilterRule(filter ChannelFilter, channels []*types.IdentifiedCh
 }
 
 // relayUnrelayedPacketsAndAcks will relay all the pending packets and acknowledgements on both the src and dst chains.
-func relayUnrelayedPacketsAndAcks(ctx context.Context, log *zap.Logger, src, dst *Chain, maxTxSize, maxMsgLength uint64, srcChannel *ActiveChannel, channels chan<- *ActiveChannel) {
+func relayUnrelayedPacketsAndAcks(ctx context.Context, log *zap.Logger, wg *sync.WaitGroup, src, dst *Chain, maxTxSize, maxMsgLength uint64, srcChannel *ActiveChannel, channels chan<- *ActiveChannel) {
 	// make goroutine signal its death, whether it's a panic or a return
 	defer func() {
+		wg.Done()
 		channels <- srcChannel
```

Comment:
One interesting thing I noticed here -- we signal to `wg.Done()` first, which is good because that will unblock the main loop's `wg.Wait()` call. However, it is possible that the following `channels <- srcChannel` send may block if the channel is full.

Up on L33, we initialize `channels` to capacity 10. It is unlikely right now that we will have 10 channels, but I think we can pick a more meaningful value if we initialize it with capacity `len(srcChannels)`.

Reply:
Excellent point! I forgot we couldn't do this before because we initialized the channel outside of the loop before we queried the channels but, now that we perform the query outside of the loop too this would be a nice change. Will follow up with a PR to make this change.